From 5994e984bd047bbe6f233af4d5fab38b4c92c5c3 Mon Sep 17 00:00:00 2001
From: Quanyang Wang <quanyang.wang@windriver.com>
Date: Thu, 11 Apr 2019 14:18:51 +0800
Subject: [PATCH 19/30] sound: imx-cdnhdmi: replace codec to component

Because of upstream commit 999f7f5af8eb ("ASoC: remove Codec related code"),
the "struct snd_soc_codec" should be replaced with "struct snd_soc_component".

Signed-off-by: Quanyang Wang <quanyang.wang@windriver.com>
---
 sound/soc/fsl/imx-cdnhdmi.c | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/sound/soc/fsl/imx-cdnhdmi.c b/sound/soc/fsl/imx-cdnhdmi.c
index 8dc9953a26a0..d0c9f1b8e333 100644
--- a/sound/soc/fsl/imx-cdnhdmi.c
+++ b/sound/soc/fsl/imx-cdnhdmi.c
@@ -148,8 +148,8 @@ static int get_edid_info(struct snd_soc_card *card)
 	struct snd_soc_pcm_runtime *rtd = list_first_entry(
 		&card->rtd_list, struct snd_soc_pcm_runtime, list);
 	struct snd_soc_dai *codec_dai = rtd->codec_dai;
-	struct snd_soc_codec *codec = codec_dai->codec;
-	struct hdmi_codec_pdata *hcd = codec->dev->platform_data;
+	struct snd_soc_component *component = codec_dai->component;
+	struct hdmi_codec_pdata *hcd = component->dev->platform_data;
 	struct imx_cdnhdmi_data *data = snd_soc_card_get_drvdata(card);
 	int i, j, ret;
 	const u8 *sad;
@@ -157,7 +157,7 @@ static int get_edid_info(struct snd_soc_card *card)
 	unsigned int rate_mask = 0;
 	unsigned int rate_mask_eld = 0;
 
-	ret = hcd->ops->get_eld(codec->dev->parent, hcd->data,
+	ret = hcd->ops->get_eld(component->dev->parent, hcd->data,
 					    data->eld, sizeof(data->eld));
 	sad = drm_eld_sad(data->eld);
 	if (sad) {
@@ -281,12 +281,12 @@ static int get_edid_rx_info(struct snd_soc_card *card)
 	struct snd_soc_pcm_runtime *rtd = list_first_entry(
 		&card->rtd_list, struct snd_soc_pcm_runtime, list);
 	struct snd_soc_dai *codec_dai = rtd->codec_dai;
-	struct snd_soc_codec *codec = codec_dai->codec;
-	struct hdmi_codec_pdata *hcd = codec->dev->platform_data;
+	struct snd_soc_component *component = codec_dai->component;
+	struct hdmi_codec_pdata *hcd = component->dev->platform_data;
 	struct imx_cdnhdmi_data *data = snd_soc_card_get_drvdata(card);
 	int ret;
 
-	ret = hcd->ops->get_eld(codec->dev->parent, hcd->data,
+	ret = hcd->ops->get_eld(component->dev->parent, hcd->data,
 					    data->eld, sizeof(data->eld));
 
 	if (ret)
-- 
2.17.1

